x-pack/plugins/enterprise_search/public/applications/__mocks__/wait_for_promises.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * If you created an immediately resolved promise, you can use this to wait for it to resolve without a reference.
+ *
+ * http.get.mockReturnValueOnce(Promise.resolve({ analyticsUnavailable: true }));
+ * AnalyticsLogic.actions.loadQueryData('some-query');
+ * 
+ * await waitForPromises();
+        
+ * expect(AnalyticsLogic.actions.onAnalyticsUnavailable).toHaveBeenCalled();
+ *
+ */
+export function waitForPromises() {
+  let promiseResolve: (value: unknown) => void;
+  const promise = new Promise((resolve) => (promiseResolve = resolve));
+  setTimeout(() => promiseResolve(undefined));
```

Comment:
same-ish question as above? Why do we need the undefined?

```suggestion
  setTimeout(() => promiseResolve());
```